ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java

```diff
@@ -280,6 +280,7 @@ public int monitorExecution() {
             // best effort
           }
           console.printError("Execution has failed. stack trace: " + ExceptionUtils.getStackTrace(e));
+          diagnostics.append(e.getMessage());
```

Comment:
is that an intentional change? just asking

Reply:
while I find this very useful, a bit confused why it's needed, is there a chance you can repro a failure to see how is the error messages are handled?
I mean:
https://github.com/apache/hive/blob/cf83d4751271622a9a700c6f2330dbfff38801d2/ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java#L296
so in the finally block, the final dagStatus (given by the tez am) also contains diagnostics, which is then printed to the console and also added to diagnostics
so this change assumes that there is an error which is swallowed here and is not part of the final dagStatus
if that's the case, it's okay append to the diagnostics, we need to just understand what happened

Reply:
The code you pointed is about copying messages from the `DAGStatus` instance. However in this special test case  this instance is `null` because the `LockException` is thrown before pulling the status.
https://github.com/apache/hive/blob/cf83d4751271622a9a700c6f2330dbfff38801d2/ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java#L171

https://github.com/apache/hive/blob/cf83d4751271622a9a700c6f2330dbfff38801d2/ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java#L183

So it seems that heartbeat related checks are handled independently from the tez jobs status and the response coming from the first one is not fed into diagnostics.

Reply:
I see, this is an edge where we don't even reach real dag monitoring then
not sure where the diagnostics object is used, but does it make sense to use the full trace, like:
```
         String reportedException = "Execution has failed, stack trace: " + ExceptionUtils.getStackTrace(e);
          console.printError(reportedException);
          diagnostics.append(reportedException);
```